unstructured/partition/pdf.py

```diff
@@ -1002,8 +1106,9 @@ def get_word_bounding_box_from_element(
 
                 # TODO(klaijan) - isalnum() only works with A-Z, a-z and 0-9
                 # will need to switch to some pattern matching once we support more languages
-                if index == 0:
+                if not set_alnum:
                     isalnum = char.isalnum()
+                    set_alnum = True
 
                 if char.isalnum() != isalnum:
```

Comment:
maybe think about this problem from a different angle? What does `isalnum` 's intention? It seems that if a `char`'s check is not the same as the last check we add a word? So on first pass we always expect the next condition to pass at line 1113 and word would be not empty after that. 
So in other words if word is empty we know we don't need to check at line 1113 so we can replace the highlighted lines with

``` 
if not word:
    isalnum = char.isalnum()
if word and char.isalnum() != isalnum:
```